lib/admin.php

```diff
@@ -338,5 +338,73 @@ function woocommerce_embed_page_header() {
 	</div>
 	<?php
 }
-
 add_action( 'in_admin_header', 'woocommerce_embed_page_header' );
+
+/**
+ * Registers WooCommerce specific user data to the Wordpress user API.
+ */
+function wc_admin_register_user_data() {
+	register_rest_field(
+		'user',
+		'woocommerce_meta',
+		array(
+			'get_callback'    => 'wc_admin_get_user_data_values',
+			'update_callback' => 'wc_admin_update_user_data_values',
+			'schema' => null,
+		)
+	);
+}
+ add_action( 'rest_api_init', 'wc_admin_register_user_data' );
+
+ /**
+  * We store some WooCommerce specific user meta attached to users endpoint,
+  * so that we can track certain preferences or values such as the inbox activity panel last open time.
+  * Additional fields can be added in the function below, and then used via wc-admin's currentUser data.
+  *
+  * @return array Fields to expose over the WP user endpoint.
+  */
+function wc_admin_get_user_data_fields() {
+	$user_data_fields = array(
+		'revenue_report_columns',
+	);
+
+	return apply_filters( 'wc_admin_get_user_data_fields', $user_data_fields );
+}
+
+ /**
+  * For all the registered user data fields (  wc_admin_get_user_data_fields ), fetch the data
+  * for returning via the REST API.
+  *
+  * @param WP_User $user Current user.
+  */
+function wc_admin_get_user_data_values( $user ) {
+	$values = array();
+	foreach ( wc_admin_get_user_data_fields() as $field ) {
+		$values[ $field ] = get_user_meta( $user['id'], 'woocommerce_meta_' . $field, true );
```

Comment:
Could you see anyone using this to get user meta not prefixed with `woocommerce_meta`?

Was thinking of using this as a 3rd party developer and it might be nice if we could add in other user meta fields:

```php
	$user_data_fields = array(
		'revenue_report_columns' => 'woocommerce_meta_revenue_report_columns',
         'my_custom_user_field' => 'other_plugin_prefix_meta_field',
	);
```

Reply:
Maybe we should change the prefix to `wc_admin` instead? That describes the usage and still namespaces it a bit. 3rd parties can still extend via the filter.

Reply:
@coderkevin Sounds like a good idea to me!

Reply:
In d40ed87, I changed the prefix to `wc_admin_`, but kept the user meta as `woocommerce_meta`. Now there's room for other stuff there, but anything that we use in wc_admin will be prefixed appropriately.